File: Bag_ADT/Python/Bag.py

class Bag:

    def __init__(self):
        self.bagContainer = []


    def add(self, val):
        self.bagContainer.append(val)
    

    def contains(self, val):
        
        for i in self.bagContainer:
            if i == val:
                return True
            
        return False

    
    def remove(self, val):
        for i in self.bagContainer:
            if i == val:
                self.bagContainer.remove(i)

File: Bag_ADT/Java/Bag.java
package mann.mike.bagContainer;
import java.util.*;


public class Bag 
{
	private ArrayList bagContainer = new ArrayList();
	
	
	
	public void add(int val)
	{
		this.bagContainer.add(val);
	}
	
	
	
	public boolean contains(int val)
	{
		return this.bagContainer.contains(val);
	}
	
	
	public void remove(int val)
	{
		for (int i = 0; i < this.bagContainer.size(); ++i)
		{	
			
			if ((int)this.bagContainer.get(i) == val)
			{
				this.bagContainer.remove(i);
				return;
			}
		}
		
	}
		
}
